src/sema/expression/assign.rs

```diff
@@ -13,6 +13,22 @@ use solang_parser::diagnostics::Diagnostic;
 use solang_parser::pt;
 use solang_parser::pt::CodeLocation;
 
+///Helper functions
+fn check_immutable(context: &ExprContext, ns: &Namespace, loc: &pt::Loc, immutable: &bool, diagnostics: &mut Vec<Diagnostic>) -> Result<(), ()> {
+    if *immutable {
+        if let Some(function_no) = context.function_no {
+            if !ns.functions[function_no].is_constructor() {
+                diagnostics.push(Diagnostic::error(
+                    *loc,
+                    "cannot assign to immutable outside of constructor".to_string(),
+                ));
+                return Err(());
+            }
+        }
+    }
+    Ok(())
+}
```

Comment:
Actually adding this diagnostic should not be fatal and cause resolving to fail. Please make this function return nothing.

Reply:
Hi @seanyoung ! Thanks for the review :)
I have fixed the code, according to your suggestions.
If the rest of it is okay, I shall start working on the other refactorings.